mycore-mods/src/main/java/org/mycore/mods/classification/MCRClassificationMappingEventHandler.java

```diff
@@ -96,9 +96,9 @@ private static List<Map.Entry<MCRCategoryID, MCRCategoryID>> getXMappings(MCRCat
             final MCRCategoryDAO dao = MCRCategoryDAOFactory.getInstance();
             String label = labelOptional.get().getText();
             return Stream.of(label.split("\\s"))
-                .map(categIdString -> categIdString.split(":"))
-                .filter(categIdArr -> categIdArr != null && categIdArr.length > 1)
-                .map(categIdArr -> new MCRCategoryID(categIdArr[0], categIdArr[1]))
+                .filter(MCRCategoryID::isValid)
```

Comment:
Code is not equivalent. Not sure what the actually expected behavior is. Existing code might (also) not handle edge cases correctly.

Examples:

| Input | Existing code     |  This PR  |
| :---:   | :---: | :---: |
| `foo:bar:baz` | List containing `foo:bar`   | Empty list   |
| `:bar` | Exception   | Empty list   |
| `:bar:` | Exception   | Empty list   |
| `:bar:baz` | Exception   | List containing `foo:bar`   |